clients/python-wrapper/tests/integration/test_object.py

```diff
@@ -358,3 +390,34 @@ def test_write_read_csv(setup_repo):
             assert row == columns
         else:
             assert row == sample_data[i - 1]
+
+
+def test_writer_with_failure(setup_repo):
+    _, repo = setup_repo
+    obj = repo.branch("main").object("test_object")
+
+    try:
+        with obj.writer() as writer:
+            writer.write("Hello World!")
+            raise ValueError("Bad thing happened")
+    except ValueError:
+        pass
+
+    # Verify writer is closed
+    with expect_exception_context(ValueError):
```

Comment:
Is there a more specific exception to assert?

Reply:
I am mimicking the default behavior of the fd received from Python's built-in `open`. I'll add a message validation to the tests as well